### PR TITLE
State cache switching writes to reads during commit

### PR DESCRIPTION
**Pull request by AlexeyAkhunov**
This cache is the extension of the "batch" that is currently used for the "Execution" stage. The main difference is that upon the commit, the cache does not clear the committed entries, but marks them as "read" entries. Tested only on `call_traces` stage so far, with only "virtual" commits, will do some tests for the `execution` stage on HDD.
This PR also cleans up the previous prototype of the shared stage, as well as old code used to build state visualisations

### CODE REVIEW

File: turbo/shards/state_cache.go [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+
+// LRU state cache consists of two structures - B-Tree and binary heap
+// Every element is marked either as Read, Updated, or Deleted via flags
+
+const (
+	ModifiedFlag uint16 = 1
+	DeletedFlag  uint16 = 2
+)
+
+//nolint:maligned
```
Comment (AskAlexSharov) [inline turbo/shards/state_cache.go:22]: To satisfy this linter: move fixed size fields to top of struct declaration. 
Move down fields of type `Account` and `[]byte` - their size is unknown.

File: turbo/shards/state_cache.go [hunk 2]
```diff
@@ -0,0 +1,957 @@
...
+	case *StorageItem:
+		c := bytes.Compare(ci.addrHash.Bytes(), i.addrHash.Bytes())
+		if c == 0 {
+			// Code comes before storage items
+			if ci.incarnation == i.incarnation {
+				return true
+			}
+			return ci.incarnation < i.incarnation
+		}
+		return c < 0
```
Comment (AskAlexSharov) [inline turbo/shards/state_cache.go:340]: you don't compare storage hash, will it lead to: "2 updates of 1 storage record will produce 2 different records in tree"?

Comment (AlexeyAkhunov) [reply turbo/shards/state_cache.go:340]: This is where `CodeItem` is compared with`StorageItem`, not two `StorageItem`s compared with each others. Therefore, `CodeItem` does not have location hash to compare with. Above, there is a function `Less` for `StorageItem` that handles the case you are thinking of.

File: turbo/shards/state_cache.go (hunk 1 above)
Comment (AlexeyAkhunov) [reply turbo/shards/state_cache.go]: Thank you, fixed

**Pull request merged**